### PR TITLE
Upload build script, ConsolidateArtifacts.bat, and update .gitmodules

### PR DESCRIPTION
**Pull request by veraconffx**
1. Updated the .gitmoduels so that all the submodules link to the qt submodules.
Please run "git submodule sync" to maker sure it is synchronized correctly.
2. Upload the build script: the build script assumes that python, perl, and jom are already added to PATH.
The script will build qt static library for winrt-x64-msvc2015. The output library will be in Qt\qtbase\
3. Added ConsolidateArtifacts.bat. The script will copy minimum set of the library to .\artifacts and remove all other intermediate files.

### CODE REVIEW

File: BUILD_QT.bat [hunk 1]
```diff
@@ -0,0 +1,23 @@
+rem The script assumed that perl, jom and python are already set in PATH. The output dir will be Qt\[BUILD_ID]
```
Comment (krunal-patel-fl4re) [inline BUILD_QT.bat]: I will put in some ms-dos magic stuff here

File: BUILD_QT.bat [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+SET QMAKESPEC=winrt-x64-msvc2015
+
+call "%VS140COMNTOOLS%..\..\VC\vcvarsall.bat" x64
+
+SET PATH=%SRC_DIR%\qtbase\bin;%SRC_DIR%\gnuwin32\bin;%PATH%
+SET PATH=%SRC_DIR%\qtrepotools\bin;%PATH%
+
+
+@ECHO ON
+git submodule update --init --recursive
```
Comment (krunal-patel-fl4re) [inline BUILD_QT.bat]: this is a build script, so git commands should already have been executed by now.  This line needs to be removed.  I will handle

File: BUILD_QT.bat (hunk 1 above)
Comment (mbarnes-sb) [inline BUILD_QT.bat]: I assume you mean `errorlevel` checking?

**Comment by mbarnes-sb** [comment]
The Batch scripts need `errorlevel` checking after each command.

File: ConsolidateArtifacts.bat [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+if exist "%destination%" rmdir "%destination%" /s/q
+
+echo "Packaging library"
+xcopy "%static_lib_path%\include"  "%destination%\include" /y/s/q/i
+xcopy "%static_lib_path%\lib"  "%destination%\lib" /y/s/q/i
+xcopy "%static_lib_path%\bin"  "%destination%\bin" /y/s/q/i
+xcopy "%static_lib_path%\plugins"  "%destination%\plugins" /y/s/q/i
+
+REM remove all intermediate files in the original library folder
+rmdir %lib_path% /s /q
```
Comment (mbarnes-sb) [inline ConsolidateArtifacts.bat]: protect path with quotes

File: ConsolidateArtifacts.bat [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+rmdir "lib\cmake" /s /q
+if !errorlevel! neq 0 exit /b !errorlevel!
+del *.prl /s /q
+if !errorlevel! neq 0 exit /b !errorlevel!
+del *.pdb /s /q
+if !errorlevel! neq 0 exit /b !errorlevel!
+
+pushd bin
+REM delete all applications except for necessary ones.
+for %i in (*.*) do if not "%i"=="moc.exe" if not "%i"=="rcc.exe" if not "%i"=="uic.exe" del /q "%i"
```
Comment (krunal-patel-fl4re) [inline ConsolidateArtifacts.bat]: this is causing a problem http://50.250.211.65:7070/job/build-qt5-on-windows/14/console

**Pull request merged**